IEXSharp/Service/Cloud/StockFundamentals/StockFundamentalsService.cs

```diff
@@ -148,10 +154,16 @@ public async Task<IEXResponse<IEnumerable<SplitBasicResponse>>> SplitsBasicAsync
 			{
 				var splitResponse = await executor.ExecuteAsync<SplitBasicResponse>(urlPattern, pathNvc, qsb);
 
-				return new IEXResponse<IEnumerable<SplitBasicResponse>>()
-				{
-					Data = new List<SplitBasicResponse> { splitResponse.Data }
-				};
+				return splitResponse != null
+					? new IEXResponse<IEnumerable<SplitBasicResponse>>()
+					{
+						ErrorMessage = splitResponse.ErrorMessage,
+						Data = new List<SplitBasicResponse> { splitResponse.Data }
```

Comment:
Is this right here - it might just be because it's getting late, but you're populating an error message and then adding the split response data in the same block, and then in the next you're returning a freshly initialised list. Would it not be: Error + new list and in the other block the list with the split response data?

Reply:
The reason that the Data is in the first one is bc there is a != when comparing with null (rather than ==).  
You're technically right about including Error even with the new List though. It would never happen in the current code where the new List is paired w/ an error, since ExecuteAsync() only returns null when there is no error. But I will include a passthrough for the Error just in case things change with ExecuteAsync() in the future.

Reply:
Actually, NM, if we change the code to this:
```c#
				return splitResponse != null
					? new IEXResponse<IEnumerable<SplitBasicResponse>>()
					{
						ErrorMessage = splitResponse.ErrorMessage,
						Data = new List<SplitBasicResponse> { splitResponse.Data }
					}
					: new IEXResponse<IEnumerable<SplitBasicResponse>>()
					{
						ErrorMessage = splitResponse.ErrorMessage,
						Data = new List<SplitBasicResponse>()
					};
```
It would cause a NRE on at the third line from the end as it tries to get an ErrorMessage out of a null splitResponse.

Reply:
Yep, thanks! It’s too late for me to be reading code! Haha

Reply:
Lol, no problem. Get a good night's sleep and we can look at the code tomm.